origin-js/src/resources/marketplace.js

```diff
@@ -108,25 +109,34 @@ class Marketplace {
     const ipfsHash = this.contractService.getIpfsHashFromBytes32(
       chainOffer.ipfsHash
     )
+
     const ipfsOffer = await this.ipfsDataStore.load(OFFER_DATA_TYPE, ipfsHash)
 
     // validate offers awaiting approval
     if (chainOffer.status === 'created') {
       const listing = await this.getListing(listingId)
+      const listingCommision =
+          listing.commission && typeof listing.commission === 'object' ?
+            await this.contractService.moneyToUnits(listing.commission) :
+            '0'
+
+      if (listing.type === 'unit') {
+        // TODO(John) - there is currently no way to know the currency of a fractional listing.
+        // We probably need to add a required "currency" field to the listing schema and write a check here
+        // to make sure the chainOffer and the listing have the same currency
+        const listingCurrency = listing.price && listing.price.currency
+        const listingPrice = await this.contractService.moneyToUnits(listing.price)
+        const currencies = await this.contractService.currencies()
+        const currency = listingCurrency && currencies[listingCurrency]
+        const currencyAddress = currency && currency.address
+
+        if (currencyAddress && currencyAddress !== chainOffer.currency) {
+          throw new Error('Invalid offer: currency does not match listing')
+        }
 
-      const listingCurrency = listing.price && listing.price.currency
-      const listingPrice = await this.contractService.moneyToUnits(listing.price)
-      const listingCommision = await this.contractService.moneyToUnits(listing.commission)
-      const currencies = await this.contractService.currencies()
-      const currency = currencies[listingCurrency]
-      const currencyAddress = currency && currency.address
-
-      if (currencyAddress !== chainOffer.currency) {
-        throw new Error('Invalid offer: currency does not match listing')
-      }
-
-      if (listingPrice > chainOffer.value) {
-        throw new Error('Invalid offer: insufficient offer amount for listing')
+        if (listingPrice && listingPrice > chainOffer.value) {
```

Comment:
Since moneyToUnits returns a string, aren't we comparing strings here whereas we should be comparing numbers ?
I'm thinking moneyToUnits should always return a BigNumber and then we use BigNumber.isGreaterThan

Reply:
Good catch! Looks like we've been comparing strings for awhile... I changed it so we're now comparing numbers, but didn't make the change to return a BigNumber yet b/c that seems like its own task that would require a bunch of changes and testing... but I think this works for now - yeah?